apps/shared/avifpng.c

```diff
@@ -17,6 +20,134 @@ typedef png_bytep png_iccp_datap;
 typedef png_charp png_iccp_datap;
 #endif
 
+// Converts a hexadecimal string which contains 2-byte character representations of hexadecimal values to raw data.
+// hexString may contain values consisting of [A-F][a-f][0-9] in pairs, e.g., 7af2..., separated by any number of newlines.
+// On success the bytes are filled and AVIF_TRUE is returned.
+// AVIF_FALSE is returned if fewer than expectedLength hexadecimal pairs are converted.
+static avifBool avifHexStringToBytes(const char * hexString, size_t expectedLength, avifRWData * bytes)
+{
+    // Preprocess the input hexString by removing a tag commonly added at encoding, if present.
+    // HEIF specification ISO-23008 section A.2.1 allows including and excluding it from AVIF files.
+    // The PNG 1.5 extension mentions the omission of this header for the modern standard eXIf chunk.
+    const char tagExif00[] = "457869660000"; // "Exif\0\0" tag encoded as a hexadecimal string.
+    const size_t tagExif00Len = 6 * 2;
+    if ((expectedLength >= tagExif00Len) && !memcmp(hexString, tagExif00, tagExif00Len)) {
+        hexString += tagExif00Len;
+        expectedLength -= tagExif00Len;
+    }
```

Comment:
IMPORTANT: The variables `expectedLength` and `tagExif00Len` have different units. `expectedLength` is in converted bytes, whereas `tagExif00Len` is in hex digits. It seems that this if statement should read:

```
    if ((expectedLength >= tagExif00Len / 2) && !memcmp(hexString, tagExif00, tagExif00Len)) {
        hexString += tagExif00Len;
        expectedLength -= tagExif00Len / 2;
    }
```

Since you have apparently tested this code with Exif metadata containing this tag, I may have misunderstood this code.

Note: The use of `memcmp` (instead of `strcmp`) here requires `hexString` to have at least `tagExif00Len` characters. I verified this condition is ensured by the check `(unsigned long)expectedLength > (hexPayloadMaxLength / 2)` at line 96.

Nit: Should we allow initial `'\n'` characters before "457869660000"?